docs/dxp/7.x/en/liferay-internals/extending-liferay/overriding-osgi-services/reconfiguring-components-to-use-your-osgi-service/resources/liferay-m1t1.zip/m1t1-custom-impl/src/main/java/com/acme/m1t1/internal/M1T1CustomImpl.java

```diff
@@ -0,0 +1,35 @@
+package com.acme.m1t1.internal;
+
+import com.acme.m1t1.M1T1;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+@Component(immediate = true, service = M1T1.class)
```

Comment:
I used the `service.ranking:Integer` component property to rank this service higher than the existing service:

```java
@Component(property = {"service.ranking:Integer=100"}, service = S1J6.class)
```